### ISSUE TITLE
Question about the AGENDA dataset

### ISSUE THREAD

**Issue by zhongluwang** [rikdz/GraphWriter#4]
What's the meaning of the last column in _data/preprocessed.train.tsv_, such as _6 1 4 5 8 7 -1 0 3 7 -1 2 7 -1_? Can you introduce more about the process？

**Comment by rikdz** [comment]
Please ignore this column

**Comment by agrawal-rohit** [comment]
Similar doubt, what does the third column in _data/preprocessed.train.tsv_ mean. Example value in the column: 17 5 16 ; 1 0 11 ; 2 0 11 ; 2 0 7 ; 1 0 6 ; 16 4 12 ; 17 5 12 ; 3 1 0 ; 14 1 2

Does this describe relations between the entities in an encoded form?

**Comment by rikdz** [comment]
Thanks for your interest. 
This is the actual graph, encoded as <head, relation, tail> triples divided by ';' 
The indexes for head and tail reference the entities list (the second column in the dataset), and the relations are listed in data/relations.vocab. This format was processed from the data in data/unprocessed.tar.gz by some auxiliary formatting tools which are not included in the repo. 

Please let me know if you have any other questions!

**Comment by GreatlakesSreeharsha** [comment]
The unprocessed datasets already contains the "relations":graph<head,relation,tail>, can you please elaborate what algorithm was used to create this relation. Was it based on SicIE only or any other technique?

**Comment by rikdz** [comment]
@GreatlakesSreeharsha this was done with DygIE https://github.com/luanyi/DyGIE

**Comment by anandhperumal** [comment]
@rikdz I should say this, I'm impressed by your work, thank you for sharing the code with us. I trying to use my own knowledge graph but I don't understand what is the last column in preproccessed.train.tsv indicates ``such as 6 1 4 5 8 7 -1 0 3 7 -1 2 7 -1? ``` As you replied @zhongluwang to ignore this but isn't that an important column when we want to train a model? Since that column is being used internally.

**Comment by pratik2358** [comment]
This is a fascinating work!
@rikdz I am unable to understand the third and the last column. Would you please explain those for me?

**Comment by pratik2358** [comment]
@rikdz and if I want to give my custom inputs (titles and graphs), how would I be able to do that?

**Comment by rikdz** [comment]
You can process your data to a tab-separated list with the following fields:

title, entities, entity type, graph, target, ordering

"entities" is a semi-colon delimited list of entities
"entity type" is a space delimited list of one type token associated with each entity
"graph" is a semi-colon delimited list of graph triples, where the head and tail are indexes of the entities in the "entities" list and the relation is the index of the relation in the data/relations.vocab file
"target" is your target text with entities replaced by placeholders indicating the entity type and its index in the "entities" list
ordering is not used, but you may need to put a placeholder there to get the code to work as is. Alternately, you can modify the code to ignore this field 

hope this helps

**Comment by pratik2358** [comment]
Thanks a lot sir. This makes things a lot easier for me.

**Comment by wangobango** [comment]
Hello. Can you tell us what the last column in preprocessed data was supposed to mean? Will the model train well if I just ignore the last column or fill it with random data ?

**Comment by rikdz** [comment]
Yes, the last column could be filled with random data and it should not
affect training.
